core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -546,10 +565,52 @@ private RelationPlan createInsertPlan(Analysis analysis, Insert insertStatement)
         TableHandle tableHandle = insert.getTarget();
         Query query = insertStatement.getQuery();
         Optional<TableLayout> newTableLayout = insert.getNewTableLayout();
-        return getInsertPlan(analysis, insert.getTable(), query, tableHandle, insert.getColumns(), newTableLayout, Optional.empty());
+        return getInsertPlan(analysis, Optional.of(insert.getTable()), query, tableHandle, insert.getColumns(), newTableLayout, Optional.empty());
+    }
+
+    private RelationPlan createMaterializedViewCreationPlan(Analysis analysis)
+    {
+        checkState(analysis.getCreateMaterializedView().isPresent(), "CreateMaterializedViewAnalysis handle is missing");
+        Analysis.CreateMaterializedViewAnalysis create = analysis.getCreateMaterializedView().get();
+        createMaterializedView(create);
```

Comment:
this will bypass target table analysis that happens during `refresh` analysis.

Think about how we would need to change SPI in order to perform similar analysis for `create MV with data` and `refresh MV`.

Reply:
Currently, there are two checks below are performed during refresh MV analysis.
1. Verify if table handle for storage table exists
2. Column types of existing storage table is same as types derived from MV definition (query)

The first checking is also performed in this PR, but it is valid only if `createMaterializedView` is called here. In the upcoming PR, I am going to change in the way that `createMaterializedView` is handled inside a connector. Then this checking doesn't make sense for `Create MV with data` case.
In my opinion, the second checking is not meaningful to `Create MV with data` case if creating MV metadata is handled inside a connector.